pkg/credentialprovider/plugin/plugin.go

```diff
@@ -269,6 +289,15 @@ func (p *pluginProvider) isImageAllowed(image string) bool {
 
 // getCachedCredentials returns a credentialprovider.DockerConfig if cached from the plugin.
 func (p *pluginProvider) getCachedCredentials(image string) (credentialprovider.DockerConfig, bool, error) {
+
+	// NewExpirationCache purges expired entries when List() is called
```

Comment:
Seems a little odd to be relying on the behavior of List() to purge expired entries, wondering if there's another method we can call to make this more obvious

Reply:
agreed but i could not find any other option, mostly this cache used as version cache in k8s codebase which I assume does not required to be purge :) and the intree credential provider plugins also not using any mechanism to purge. 
The cache purges expired entry on `get` also but only the requested one, so there is a chance if a key is never  requested and expired would never gets removed.